config/database.pgsql

```diff
@@ -68,6 +68,12 @@ CREATE TABLE IF NOT EXISTS user_available_plants(
 );
 
 
+CREATE TABLE IF NOT EXISTS user_garden_access(
+    garden_access BIGINT,
+    garden_owner BIGINT
+);
```

Comment:
You should give this a primary key so that a user can't be added twice, so that you don't need to SELECT before adding but rather you'd just need to catch `asyncpg.UniqueViolationError`.

Reply:
Problem is it's a many-to-many mapping. A user can exist several times. If garden_access is primary key, then that user can only be granted access to one garden. If garden_owner is the primary key, then the user can only grant one person a key.

Reply:
Compound primary key - `(garden_owner, garden_access)`. Each pair of those can only exist once.